### PR TITLE
Marci refactor

### PR DESCRIPTION
**Pull request by PMarci**

### CODE REVIEW

File: src/test/java/com/greenfoxacademy/controllers/ContactControllerTest.java [hunk 1]
```diff
@@ -252,6 +252,40 @@ public void badRequestPut() throws Exception {
...
+        int id = 1;
+        TestContact testContact = new TestContact("Pista bá", "Fúr, farag", 1L);
+        String validTestJson = createTestJson(testContact);
+
+        mockMvc.perform(post("/contacts")
+                .header("session_token", "abcde")
+                .header("Origin", "https://lasers-cornubite-konnekt.herokuapp.com")
+                .contentType(MediaType.APPLICATION_JSON).content(validTestJson));
+
+        mockMvc.perform(get("/contact/{id}", id)
```
Comment (adamgyulavari) [inline src/test/java/com/greenfoxacademy/controllers/ContactControllerTest.java]: @PMarci two different tests should go in two different test methods

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: src/test/java/com/greenfoxacademy/controllers/ContactControllerTest.java (hunk 1 above)
Comment (PMarci) [reply src/test/java/com/greenfoxacademy/controllers/ContactControllerTest.java]: But the class ContactControllerTest contains only one expectation per test method... did you mean forgotcontrollertest?

Comment (adamgyulavari) [reply src/test/java/com/greenfoxacademy/controllers/ContactControllerTest.java]: you're using here the mockmvc to perform a post and a get request. these should be in different tests for those, or simply create the test contact and save it with the repo

Comment (PMarci) [reply src/test/java/com/greenfoxacademy/controllers/ContactControllerTest.java]: As of now, all tests are transactional, meaning, each transaction is rolled back after the completion of a test. So putting the creation of the test contact in a different method is not feasible. Would you consider this 
```
    Contact testcontact = new Contact(userService.findUserById(1L), "John Doe", "FOOBAR");
    contactRepository.save(testcontact);

```
a more appealing solution than the current one?
 ```
       mockMvc.perform(post("/contacts")
                .header("session_token", "abcde")
                .contentType(MediaType.APPLICATION_JSON).content(validTestJson));
```

**Pull request closed without merging**